force-app/main/default/classes/OSF_AccDetailsExtension.cls

```diff
@@ -0,0 +1,53 @@
+public with sharing class OSF_AccDetailsExtension {
+    public OSF_AccDetailsExtension(ApexPages.StandardController stdController) { }
+    
+    public String actid {get; set;}
+    public list <Contact> acc {get;set;}
+    public String searchKey {get;set;}
+    String var = ApexPages.currentPage().getParameters().get('id');
+    public Pagereference setPrimary() { 
+        //If there is primary contact make it false first
+        List<Contact> primaryct = [SELECT Id FROM Contact WHERE Is_Primary_Contact__c = true AND Account.Id = :var];
+        if(primaryct.size() > 0)  {
+            Contact a = primaryct.get(0);
+            a.Is_Primary_Contact__c = false;
+        }
+        
+        //Get Account Id and set as primary contact
+        Contact queriedContact = [SELECT Id
+                          FROM Contact 
+                          WHERE Id = :actid
+                          LIMIT 1];
+        queriedContact.Is_Primary_Contact__c = true;
+
+        update queriedContact;
+        update primaryct;
+        //When process completed close pop-up
+        displayPopup = false;
+
+        //When process complete refresh page witch associated accountId
+        PageReference retURL = new PageReference('/apex/OSF_AccountDetails?id='+var);
+        retURL.setRedirect(true);
+        return retURL;
+    }  
+    //Pop-up 
+    public boolean displayPopup {get;set;}
+    public void closePopup() 
+    {
```

Comment:
Please use single formatting in controller such as "methodName {" without new line